spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/test/domain/TradeRepository.java

```diff
@@ -19,11 +19,18 @@
 import java.util.List;
 
 import org.springframework.cloud.gcp.data.spanner.repository.SpannerRepository;
+import org.springframework.cloud.gcp.data.spanner.repository.query.Query;
+import org.springframework.data.repository.query.Param;
 
 public interface TradeRepository extends SpannerRepository<Trade> {
 
 	List<Trade> findByTraderId(String traderId);
 
 	int countByAction(String action);
 
+	@Query("SELECT * FROM trades_#{tableNameSuffix} WHERE action='#{#trade.action}'")
```

Comment:
If I understand this example correct, you are actually creating a new query String with the SpEL replaced by its evaluated value. So `action='#{#trade.action}'` turns into action='BUY'

If this understanding is correct this might turn into an attack vector for SQL injection. Instead one should start with a query like `action=#{#trade.action}` (no quotes) and turn it into `action=@__synthetic__1` and bind the evaluation result of the SpEL expression to that new bind parameter.

At least in SD-JPA there is only very limited support of SpELs evaluation result actually appearing in the resulting query. You can basically only inject the entity name. This allows somewhat customizing of selects without offering much attack surface for any kind of injection attacks.